### PR TITLE
[Cleanup] Remove unused plasma* related stuff: 10.2/N

### PR DESCRIPTION
**Pull request by suchenzang**
[ Note: will rebase onto main after https://github.com/facebookresearch/metaseq/pull/366 is merged]

Removed:
* `use_plasma_view` flag
* as a result, `split_path` and `plasma_paths` become unused in `TokenBlockDataset`
* `PlasmaView` class

### CODE REVIEW

File: metaseq/data/token_block_dataset.py [hunk 1]
```diff
@@ -62,31 +57,13 @@ def __init__(
...
+
+        self._slice_indices = plasma_utils.PlasmaArray(slice_indices)
+        self._sizes = plasma_utils.PlasmaArray(_sizes)
+        self._block_to_dataset_index = plasma_utils.PlasmaArray(block_to_dataset_index)
 
     @staticmethod
-    def _build_slice_indices(
-        sizes, break_mode, document_sep_len, block_size
-    ) -> Tuple[np.ndarray]:
+    def _build_slice_indices(sizes, break_mode, document_sep_len, block_size):
```
Comment (ruanslv) [inline metaseq/data/token_block_dataset.py:66]: We don't want the type hint? This seems unrelated

Comment (suchenzang) [reply metaseq/data/token_block_dataset.py:66]: type hint was wrong

**Comment by suchenzang** [comment]
Dang, forgot our squash merges kills the ability to do clean rebases. Will be closing all of these and cherry-picking to new PRs 🤦‍♀️

**Comment by suchenzang** [comment]
Cherry-picked over in https://github.com/facebookresearch/metaseq/pull/379
